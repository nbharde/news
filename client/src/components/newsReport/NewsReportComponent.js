import React from "react";

const NewsReportComponent = ({article}) => {
    return(
        <>
        <div className="article" data-testid="headline_testId">
            <div className="image">
                <img src={article.urlToImage} alt="News" height="100" width="130"/>
            </div>
            <div className="content">
                <div className="title">{article.title}</div>
                <div className="description">{article.description}</div>
            </div>
        </div>
        </>
    );
}

export default NewsReportComponent