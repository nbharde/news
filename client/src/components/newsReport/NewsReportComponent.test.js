import React from "react";
import { render } from "@testing-library/react";
import '@testing-library/jest-dom/extend-expect';

import NewsReportComponent from "./NewsReportComponent";

const article = {
    title: "COVID-19: UK reports",
    description: "Description",
    urlToImage: ""
}

describe("Testing NewsReportComponent", () => {
    it("renders a component", () => {
        const { queryByTestId } = render(<NewsReportComponent article={article}/>);
        expect(queryByTestId("headline_testId")).not.toBeNull();
    });

    it("matches snapshot", () => {
        const { container } = render(<NewsReportComponent article={article}/>);
        expect(container).toMatchSnapshot();
    });
});


