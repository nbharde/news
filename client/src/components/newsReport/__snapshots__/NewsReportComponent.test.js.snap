// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing NewsReportComponent matches snapshot 1`] = `
<div>
  <div
    class="article"
    data-testid="headline_testId"
  >
    <div
      class="image"
    >
      <img
        alt="News"
        height="100"
        src=""
        width="130"
      />
    </div>
    <div
      class="content"
    >
      <div
        class="title"
      >
        COVID-19: UK reports
      </div>
      <div
        class="description"
      >
        Description
      </div>
    </div>
  </div>
</div>
`;
